workflows/prognostic_c48_run/runtime/validate.py

```diff
@@ -0,0 +1,51 @@
+from datetime import datetime, timedelta
+from typing import Any, Mapping, Union
+
+import dacite
+import fv3config
+from .config import UserConfig, DiagnosticFileConfig
+from .diagnostics.fortran import FortranFileConfig
+
+
+class ConfigValidationError(ValueError):
+    pass
+
+
+def validate_config(config_dict: Mapping[str, Any]):
+    """Validate configuration for segmented runs. Currently just validates chunks."""
+    validate_chunks(config_dict)
+
+
+def validate_chunks(config_dict: Mapping[str, Any]):
+    """Ensure that the time chunk size evenly divides the time dimension size for all
+    output diagnostics. Raise ConfigValidationError if not."""
+    user_config = dacite.from_dict(UserConfig, config_dict)
```

Comment:
Fine for now, but it would be nice if validate_chunks took in a `UserConfig`. Ideally `dacite` would only be used once in the code base. We should think about formalizing the distinction between fv3config and python configuration better, although maybe this feature is unique in needing to know about both.

Reply:
I think it's pretty typical to need to know about the namelist for runtime configuration.

Reply:
Maybe open an issue then? It would be shame to start passing around the whole config dict more places.